includes/plugins/co-authors-plus/class-nicename-change.php

```diff
@@ -0,0 +1,301 @@
+<?php
+/**
+ * Nicename change class.
+ *
+ * @package Newspack
+ */
+
+namespace Newspack;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * This class adds some tweak to allow for a safe change of the nicename of a user.
+ *
+ * When a nicename changes it will:
+ * * Create a redirect from the old nicename to the new one.
+ * * Update the author term created by Co-Authors Plus.
+ *
+ * It will also register a new CLI command for convenience.
+ */
+class Nicename_Change {
+
+	const OLD_NICENAME_META_KEY = '_np_old_nicename';
+
+	/**
+	 * Registers the hooks.
+	 */
+	public static function init() {
+		add_action( 'profile_update', [ __CLASS__, 'profile_update' ], 10, 2 );
+		add_action( 'template_redirect', [ __CLASS__, 'old_nicename_redirect' ] );
+
+		// add a cli command to update the nicename.
+		if ( defined( 'WP_CLI' ) && WP_CLI ) {
+			\WP_CLI::add_command( 'newspack nicename-change', [ __CLASS__, 'cli_change_nicename' ] );
```

Comment:
It would be great to get synopsis on this so it's easier to see from cli what the args should look like.